rspace/src/test/scala/coop/rchain/rspace/history/HistorySpec.scala

```diff
@@ -89,6 +125,106 @@ class HistorySpec extends FlatSpec with Matchers with OptionValues with InMemory
     } yield ()
   }
 
+  //              root                                                                      root
+  //               |                                                                          |
+  //              Skip                                                                      Skip
+  //               |               delete Leaf1                                               |
+  //           PointerBlock       then insert one alone with Leaf3                      PointerBlock
+  //           /       \         ==================================>                    /         \
+  //        Skip       Skip                                                           Skip       Skip
+  //           |          |                                                             |          |
+  //        Leaf1       Leaf2                                                         Leaf3       Leaf2
+
+  "delete collapsed a branch then insert" should "work fine" in withEmptyTrie { emptyHistory =>
+    val inserts =
+      insert(hexKey("00100")) ::
+        insert(hexKey("00220")) ::
+        Nil
+    val insertsTwo =
+      delete(hexKey("00100")) ::
+        insert(hexKey("00110")) ::
+        Nil
+
+    for {
+      historyOne                    <- emptyHistory.process(inserts)
+      keyOne                        <- historyOne.find(inserts(0).key)
+      keyTwo                        <- historyOne.find(inserts(1).key)
+      historyTwo                    <- historyOne.process(insertsTwo)
+      keyOneNotExist                <- historyTwo.find(insertsTwo(0).key)
+      newInsert                     <- historyTwo.find(insertsTwo(1).key)
+      keyTwoAfterProcess            <- historyTwo.find(inserts(1).key)
+      (_, p1)                       = keyOne
+      (keyTwoValue, p2)             = keyTwo
+      (keyTwoAfterProcessValue, p3) = keyTwoAfterProcess
+      (_, p4)                       = newInsert
+      (keyOneNotExistEmpty, _)      = keyOneNotExist
+      _                             = keyTwoAfterProcessValue shouldBe keyTwoValue
+      _                             = keyOneNotExistEmpty shouldBe a[EmptyPointer.type]
+      _                             = skipShouldHaveAffix(p1.head, List(0, 0))
+      _                             = skipShouldHaveAffix(p2.head, List(0, 0))
+      _                             = skipShouldHaveAffix(p3.head, List(0, 0))
+      _                             = skipShouldHaveAffix(p1.last, List(0, 0))
+      _                             = skipShouldHaveAffix(p2.last, List(2, 0))
+      _                             = skipShouldHaveAffix(p3.last, List(2, 0))
+      _                             = skipShouldHaveAffix(p4.head, List(0, 0))
+      _                             = skipShouldHaveAffix(p4.last, List(1, 0))
+    } yield ()
+  }
+
+  "randomly insert or delete" should "return the correct result" in withEmptyTrie {
```

Comment:
This test is running 11 minutes on CI. That is too long for one unit test, all others are in matter of seconds.

We could separate these kind of tests (performance, debugging) outside of CI and run it constantly on some testing machine or when some specific part of the changes.